gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/api/ApiMongoRepositoryImpl.java

```diff
@@ -127,7 +127,12 @@ private Query buildQuery(ApiFieldExclusionFilter apiFieldExclusionFilter, ApiCri
     private Query buildQuery(ApiFieldInclusionFilter apiFieldInclusionFilter, ApiCriteria... orApiCriteria) {
         final Query query = new Query();
         if (apiFieldInclusionFilter != null) {
-            query.fields().include(apiFieldInclusionFilter.includedFields());
+            String[] fields = apiFieldInclusionFilter.includedFields();
+            if (fields.length > 0) {
```

Comment:
Can you add a short comment to explain why we need to add at least "_id" field when no other fields are provided ?

Reply:
I added it. If it's not clear enough I can still change it 😄

Reply:
I thought about a comment more like this
```suggestion
            // If there is no field to include, then the Mongo query will behave like there is no filter, and so will include all available fields. In that case we need to explicitly add the "_id" column
            if (fields.length > 0) {
```